src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.h

```diff
@@ -2,16 +2,27 @@
 #define Z_EN_TAKARAYA_H
 
 #include "global.h"
+#include "objects/object_bg/object_bg.h"
 
 struct EnTakaraya;
 
 typedef void (*EnTakarayaActionFunc)(struct EnTakaraya*, PlayState*);
 
+#define EN_TAKARAYA_GET_SWITCH_FLAG(thisx) (u8)((thisx)->params >> 0x8)
+#define TAKARAYA_GET_TREASURE_FLAG(this) (1 << (this)->actor.params)
```

Comment:
```suggestion
#define TAKARAYA_GET_TREASURE_FLAG(thisx) (1 << (thisx)->params)
```
This should take the actor instance.

Reply:
Ah sorry my bad, forgot that was the project tendency.